### PR TITLE
[IMPROVEMENT] JSON rewrite

### PR DESCRIPTION
**Pull request by FadedCoder**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/SwagLyrics/SwagLyrics-For-Spotify/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [x] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

This removes the `beautifulsoup4` dependency from the bot and uses JSON and regex to parse the lyrics from the html.

It's speed improvements are:

HTML Type | This PR | Current Method | Improvement
------------ | -------- | ------------------ | ---------------
Legacy UI | 0.014 s | 0.043 s | ~3x
React UI | 0.0031 s | 0.042 s |  ~13x

**THIS PR IS NOT READY FOR MERGE**

I have **not** tested it directly since I don't have Spotify. Please ensure it is working. Also, I didn't test with many songs.

**NOTE:** Remove `rewrite.py` once it is ready for merge.

### CODE REVIEW

**Comment by FadedCoder** [comment]
@aadibajpai fix the `.editorconfig`, it has `end_of_line=crlf`.

**Comment by aadibajpai** [comment]
> @aadibajpai fix the `.editorconfig`, it has `end_of_line=crlf`.

wontfix

File: swaglyrics/cli.py [hunk 1]
```diff
@@ -60,6 +66,23 @@ def stripper(song: str, artist: str) -> str:
...
+            if i['tag'] in ['p', 'i', 'a', 'b', 'u']:
+                lyrics += process_json_lyrics(i['children'])
+            elif i['tag'] in ['br', 'inread-ad']:
+                lyrics.append("\n")
+        elif type(i) == list:
+            lyrics += process_json_lyrics(i)
+        elif type(i) == str:
+            lyrics.append(i)
+        else:
+            raise Exception(f"Unknown type occurred! Type = {type(i)}")
```
Comment (aadibajpai) [inline swaglyrics/cli.py:82]: why not do a TypeError then?

Comment (FadedCoder) [reply swaglyrics/cli.py:82]: Idk, making it TypeError.

**Comment by aadibajpai** [comment]
I've thought about this a lot, and I feel we shouldn't go for speed at the cost of maintainability since we're not actively developing anymore so any overhead would just come back and bite us :( 

Honestly thanks so much for coming up with a significant speed improvement tho!

**Pull request closed without merging**